test/integration/targets/git/tasks/specific-revision.yml

```
@@ -70,11 +70,11 @@
 # Same as the previous test, but this time we specify which ref
 # contains the SHA1
 - name: SPECIFIC-REVISION | update to revision by specifying the refspec
-  git:
-    repo: https://github.com/ansible/ansible-examples.git
+  git: &git_ref_spec
+    repo: "{{ repo_dir }}/shallow_branches"
```

observation: pretty sure this should be pointing to the bare repo to work...
```suggestion
    repo: "{{ repo_dir }}/shallow_branches/.git"
```